### PR TITLE
Ghost as cuda package

### PR DESCRIPTION
**Pull request by jthies**
see also #7437

### CODE REVIEW

**Comment by jthies** [comment]
superseeds #7437

**Comment by adamjstewart** [comment]
I wouldn't make changes to `intel-mkl` while #7469 is in progress.

**Comment by davydden** [comment]
@adamjstewart 

> I wouldn't make changes to intel-mkl while #7469 is in progress.

these are much smaller changesets which were open before for quite a while in https://github.com/spack/spack/pull/7437

I think it's unfair to let this contribution wait until everything is tested and checked in the monstrous https://github.com/spack/spack/pull/7469

**Comment by mgsternberg** [comment]
It's perfectly fine by me to proceed here – I will carry over the change to ``.../intel-mkl/package.py`` into #7469.

**Comment by davydden** [comment]
> It's perfectly fine by me to proceed here – I will carry over the change to .../intel-mkl/package.py into #7469.

@alalazo @adamjstewart @junghans could you guys review and merge, if this looks good to you?

**Comment by jthies** [comment]
Leave a @davydden any idea why the codecov tests keep failing? Does a commit that only changes package files even affect code coverage? Thanks!

**Comment by alalazo** [comment]
>  any idea why the codecov tests keep failing? Does a commit that only changes package files even affect code coverage? 

**TLDR** Don't worry about that for package only PRs - the important bit is that Travis CI passes

**Longer explanation**: You branched off a commit that went through a scheduled nightly build, and we run unit tests on MacOS only there to avoid delays on Travis reports. The negative number you see is basically the coverage we lose dropping MacOS tests.

**Comment by jthies** [comment]
Hi @alalazo, @adamjstewart , so what about the PR, could you merge it or let me know if you don't want me to touc the Intel-mkl package, in that case I'll just add the headers property to other blas/lapack Providers. But having it provide headers seems a good idea to me anyway.

**Comment by adamjstewart** [comment]
If @mgsternberg is fine with it, I'm fine with it.
